web/js/containers/sidebar/layer.js

```diff
@@ -81,36 +82,34 @@ class Layer extends React.Component {
     }
   }
 
+  formatDisabledTitleDate = (date, period) => {
+    var dateString;
+
+    switch (period) {
+      case 'subdaily':
+        dateString = moment(util.parseDate(date)).format('DD MMMM YYYY HH:mm') + 'Z';
+        break;
+      default:
+        dateString = moment(util.parseDate(date)).format('DD MMMM YYYY');
+    }
+
+    return dateString;
+  }
+
   getDisabledTitle = (layer) => {
-    var startDate, endDate;
-    if (layer.startDate && layer.endDate) {
-      startDate = util.parseDate(layer.startDate);
-      endDate = util.parseDate(layer.endDate);
-      if (layer.period !== 'subdaily') {
-        startDate =
-          startDate.getDate() +
-          ' ' +
-          util.giveMonth(startDate) +
-          ' ' +
-          startDate.getFullYear();
-        endDate =
-          endDate.getDate() +
-          ' ' +
-          util.giveMonth(endDate) +
-          ' ' +
-          endDate.getFullYear();
-      }
+    var startDate = null; var endDate = null;
```

Comment:
No need to assign newly declared variables to `null` since they will be `undefined` until assigned some other value. `undefined` is false-y so your conditionals below will still work as intended.

Also, I'd recommend using `let` or `const` over `var` (`let` in this case).

Reply:
I had to brush up on the use of `var` vs `let` vs `const` in the realm of ES6. `let` definitely makes sense here. I'll make the change.